torch/_subclasses/fake_tensor.py

```diff
@@ -1105,6 +1165,7 @@ def __exit__(self, a, b, c):
                 torch._C._set_dispatch_mode(maybe_prev_fake_mode)
             if maybe_prev_only_lift_cpu_tensors is not None:
                 torch._C._set_only_lift_cpu_tensors(maybe_prev_only_lift_cpu_tensors)
+            self.nt_tensor_id_counter = self.nt_tensor_id_initial_count
```

Comment:
This assumes that we don't enter/exit multiple times during a tracing mode. Need to actually check this.